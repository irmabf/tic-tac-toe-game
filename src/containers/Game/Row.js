import styled from 'styled-components'

const Row = styled.div`
 &:after {
 	clear: both;
  content: "";
  display: table;
 }
`

export default Row
