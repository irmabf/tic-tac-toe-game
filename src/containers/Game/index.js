import React, { Component } from 'react'

class Game extends Component { // eslint-disable-line
	render() {
		return (
			<div>
				<div className='game-board'>
					oi
				</div>
			</div>
		)
	}
}

export default Game
