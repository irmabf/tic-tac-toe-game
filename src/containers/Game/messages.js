import { defineMessages } from 'react-intl'

export default defineMessages({
	reset: {
		id: 'tic-tac-toe.containers.Game.reset',
		defaultMessage: 'Reset'
	},
	turn: {
		id: 'tic-tac-toe.containers.Game.turn',
		defaultMessage: 'It\'s {who} turn'
	},
	your: {
		id: 'tic-tac-toe.containers.Game.your',
		defaultMessage: 'your'
	},
	computer: {
		id: 'tic-tac-toe.containers.Game.computer',
		defaultMessage: 'computer'
	},
	theWinneIs: {
		id: 'tic-tac-toe.containers.Game.theWinneIs',
		defaultMessage: 'The winner is...{winner}'
	},
	draw: {
		id: 'tic-tac-toe.containers.Game.draw',
		defaultMessage: 'It\'s a Draw'
	}
})
