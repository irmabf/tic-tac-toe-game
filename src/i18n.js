/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 */
import { addLocaleData } from 'react-intl/lib/index.es'
import enLocaleData from 'react-intl/locale-data/en'
import deLocaleData from 'react-intl/locale-data/de'

import enTranslationMessages from './translations/en.json'
import deTranslationMessages from './translations/de.json'

addLocaleData(enLocaleData)
addLocaleData(deLocaleData)

export const appLocales = [
	'en',
	'de',
]

export const formatTranslationMessages = (locale, messages) => {
	const defaultFormattedMessages = locale !== 'en'
    ? formatTranslationMessages('en', enTranslationMessages)
    : {}
	return Object.keys(messages).reduce((formattedMessages, key) => {
		const formattedMessage = !messages[key] && locale !== 'en'
      ? defaultFormattedMessages[key]
      : messages[key]
		return Object.assign(formattedMessages, { [key]: formattedMessage })
	}, {})
}

export const translationMessages = {
	en: formatTranslationMessages('en', enTranslationMessages),
	de: formatTranslationMessages('de', deTranslationMessages),
}
