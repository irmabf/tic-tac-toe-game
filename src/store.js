import { applyMiddleware, combineReducers, createStore, compose } from 'redux'
import { persistStore, autoRehydrate } from 'redux-persist'
import { REHYDRATE } from 'redux-persist/constants'
import createActionBuffer from 'redux-action-buffer'
import { createLogger } from 'redux-logger'
import thunk from 'redux-thunk'

import app from './ducks/app'

const reducers = combineReducers({
	app
})

const store = createStore(reducers, compose(
	applyMiddleware(thunk, createLogger(), createActionBuffer(REHYDRATE)),
	autoRehydrate(),
	window.devToolsExtension ? window.devToolsExtension() : (f) => f
))

persistStore(store)

export default store
