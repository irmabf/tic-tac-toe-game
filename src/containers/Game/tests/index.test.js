import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'

import LanguageProvider from '../../../containers/LanguageProvider'
import { translationMessages } from '../../../i18n'
import store from '../../../store'
import Row from '../Row'
import Game from '../index'


describe('<Game />', () => {
	it('should render 3 Rows', () => {
		const renderedComponent = mount(
			<Provider store={store}>
				<LanguageProvider messages={translationMessages}>
					<Game />
				</LanguageProvider>
			</Provider>
		)
		expect(renderedComponent.find(Row).length).toBe(3)
	})
})

