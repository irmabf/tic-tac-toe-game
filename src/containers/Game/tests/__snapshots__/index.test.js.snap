// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> <Game /> renders correctly 1`] = `
<div>
  <div
    className="Row-guwMhk gMaEAB"
  >
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
  </div>
  <div
    className="Row-guwMhk gMaEAB"
  >
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
  </div>
  <div
    className="Row-guwMhk gMaEAB"
  >
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
    <a
      className="A-jrLLFi iAHREn"
      href={undefined}
      onClick={[Function]}
    />
  </div>
</div>
`;
